repository.tf

```
@@ -30,45 +30,85 @@ resource "github_repository" "this" {
   }
 }
 
-resource "github_branch_protection" "main" {
+resource "github_repository_ruleset" "main" {
   for_each = var.repositories
 
-  repository_id                   = github_repository.this[each.key].node_id
-  pattern                         = "main"
-  enforce_admins                  = !each.value.allow_bypass_protection
-  allows_deletions                = false
-  require_signed_commits          = true
-  require_conversation_resolution = true
-
-  required_pull_request_reviews {
-    required_approving_review_count = each.value.required_approvals
-    require_code_owner_reviews      = each.value.require_code_owner_reviews
-    dismiss_stale_reviews           = true
-    require_last_push_approval      = each.value.required_approvals > 0
+  name        = format("%s-%s", each.key, "main")
+  repository  = github_repository.this[each.key].name
+  target      = "branch"
+  enforcement = "active"
+
+  conditions {
+    ref_name {
+      include = ["main"]
+      exclude = []
+    }
   }
 
-  required_status_checks {
-    strict   = true
-    contexts = each.value.required_status_checks
+  rules {
+    creation            = true
+    update              = true
+    deletion            = false
+    required_signatures = true
+    pull_request {
+      required_approving_review_count   = each.value.required_approvals
+      require_code_owner_review         = each.value.require_code_owner_reviews
+      dismiss_stale_reviews_on_push     = true
+      require_last_push_approval        = each.value.required_approvals > 0
+      required_review_thread_resolution = true // ->  require_conversation_resolution = true
```

```suggestion
      required_review_thread_resolution
```
Er ikke sikekr pa jeg forstar kommentaren. `require_conversation_resolution` er ikke gyldig prop